src/app/cart/cart.component.js

```diff
@@ -46,8 +46,12 @@ class CartController {
     } else {
       this.loading = true
     }
+    // Remember the order of the existing items in the cart
+    const orderByCode = this.cartData?.items?.map(item => item.code) || []
```

Comment:
I think you can only give one gift per frequency per designation number, so this should work. I checked on my local and it seemed to work. but I could be missing something.
Also, They stayed in order when refreshing

Reply:
"Per frequency" is the key here. This sort only looks at designation number. I was just wondering if (e.g.) the monthly and single to 0123456 will always be in that order or will they flip flop?

Reply:
The code for a monthly gift to account 0123456 is "0123456_MON". Quarterly and annually have a similar unique suffix. So the order will be preserved as long as the user doesn't have 2 monthly gifts in their cart to the same staff person. But if the user changes a gift from monthly to annual, it will move in the list. The URI can't be used as a stable unique ID because it changes when a cart item is modified.

Here is what I propose: preserve the order after edits until the frequency changes. If a gift's frequency changes, move it to the top (or bottom) because we don't know how to sort it anymore because the code changed. The main alternative I see is defining a total sort order and always using it everywhere in the UI. For example, we could sort by code, then by frequency, then by start date. The disadvantage with that is that newly-added items will appear in a random location in the list, not at the top.

Does anyone have a better idea?

Reply:
Ah yes, thanks for the reminder of the different codes. I don't think we need to do such a widespread sort. The likelihood of someone have multiple gifts to the same person and changing the frequency of one of those gifts is near zero, so that scenario doesn't need a bunch of coding around it.

Reply:
Another idea is to sort by URIs, note the URI of the item before the edit, and put the new item in the location of the original URI (since the URI of edited cart items changes). But that would require a good bit more work, and I'm not sure it's worth the effort.